File: Leet 1979/solution.py
import math
from typing import List


def find_gcd(nums: List[int]) -> int:
    """Return the Greatest Common Devisor of the smallest and largest number from nums.

    Args:
        nums (List[int]): list of ints.
    
    Returns:
        int: GCD of the smallest and largest number from nums.
    """
    mx: int = nums[0]
    mn: int = nums[0]
    
    for num in nums:
        mx = max(mx, num)
        mn = min(mn, num)

    return math.gcd(mn, mx)


def main() -> None:
    print("1979. Find Greatest Common Divisor of Array")
    
    nums: List[int] = [2, 5, 6, 9, 10]
    result: str = "pass" if find_gcd(nums) == 2 else "fail"
    print(f"Result: {result}")


if __name__ == "__main__":
    main()

File: Leet 1979/solution.cs
namespace Leet
{
    class Solution
    {
        /// <summary>
        /// Return the Greatest Common Devisor between ints a and b recursivly.
        /// </summary>
        /// <returns>gcd between a and b</returns>
        private static int Gcd(int a, int b) =>
            b == 0 ? a : Gcd(b, a % b);

        /// <summary>
        /// Return the Greatest Common Devisor of the smallest and largest number from nums.
        /// </summary>
        /// <param name="nums">array of ints</param>
        /// <returns>GCD of smallest and largest number from nums</returns>
        public static int FindGcd(int[] nums)
        {
            int mx = nums[0];
            int mn = nums[0];

            foreach (int num in nums)
            {
                mx = Math.Max(mx, num);
                mn = Math.Min(mn, num);
            }

            return Gcd(mx, mn);
        }
    }
}

class Program
{
    private static void Main()
    {
        Console.WriteLine("1979. Find Greatest Common Divisor of Array");

        int[] nums = [2, 5, 6, 9, 10];
        string result = Leet.Solution.FindGcd(nums) == 2 ? "Pass" : "Fail";
        Console.WriteLine($"Case1 should equal 2? {result}");
    }
}